### DISCUSSION TITLE
Method for comms from external application?

### DISCUSSION THREAD

**Discussion by fcwze** [NationalSecurityAgency/ghidra#4045] [Q&A]
Is there a recommended method for communicating with Ghidra from an external application?

Specifically I'm looking for something fairly light-weight that will allow me to update the current view in a particular CodeBrowser instance to match the address selected in an external application.

**Comment by ryanmkurtz** [comment]
Ghidra does not support this.  You would have to write a Ghidra extension and implement your own external communication system...definitely not light-weight.

**Reply by fcwze** [reply]
A colleague suggested a python script to run a UDP socket within a separate thread but it's not an area I'm too familiar with and I have concerns that if not careful it could end up hogging a lot of processing resources.

**Reply by astrelsky** [reply]
> A colleague suggested a python script to run a UDP socket within a separate thread but it's not an area I'm too familiar with and I have concerns that if not careful it could end up hogging a lot of processing resources. 

As long as you don't write your own udp socket implementation this shouldn't be a concern. I think TCP is more appropriate but I could be wrong (not a networking guy).
Anyway a standard implementation would be using events/interrupts so the thread would be asleep unless communicating. You should expect low resource usage from it. This is assuming your not sending/receiving gigs of data when communicating of course.


For reference Ghidra communicates with the decompiler through a pipe and uses xml for serialization. You could do something similar if you have the option of starting the python script from the java plugin.
